### PR TITLE
add group_pattern and host_filter to foreman plugin

### LINKED ISSUE #63613: foreman inventory: support group_patterns
##### SUMMARY
Please support group patterns supported by the inventory script in contrib: https://github.com/ansible/ansible/blob/devel/contrib/inventory/foreman.ini#L86

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
./lib/ansible/plugins/inventory/foreman.py

##### ADDITIONAL INFORMATION
See ./contrib/inventory/foreman.py

### PR DESCRIPTION
**Pull request by jainnikhil30**
##### SUMMARY
fixes #63613 #63611

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME
foreman plugin

### CODE REVIEW

File: lib/ansible/plugins/inventory/foreman.py [hunk 1]
```diff
@@ -270,6 +294,18 @@ def _populate(self):
                             self.display.warning("Could not set hostvar ansible_ssh_host to '%s' for the '%s' host, skipping: %s" %
                                                  (host['ip'], host_name, to_text(e)))
 
+                if self.get_option('group_patterns'):
```
Comment (AlanCoding) [inline lib/ansible/plugins/inventory/foreman.py:297]: might want to check if `self.get_option('keyed_groups')` is non-null, and if it is, might be best to throw an error or a warning. I'm not sure exactly the precedent for handling this.

**Review by jladdjr — APPROVED** [review]
LGTM!

**Comment by jladdjr** [comment]
@ares @ehelms @ekohl @xprazak2 - What's needed in order to move this forward? The awx team is trying to get this in so that the foreman plugin can be adopted for Tower 3.7.

cc @ryanpetrello @blomquisg

**Comment by jladdjr** [comment]
ready_for_review

**Comment by jainnikhil30** [comment]
@ares @ehelms @ekohl @xprazak2

Could you please check on this ? Thanks.

**Comment by wenottingham** [comment]
This probably should be closed, this isn't maintained here any more.

**Comment by ryanpetrello** [comment]
Agreed w/ @wenottingham, this doesn't belong here anymore.

**Comment by AlanCoding** [comment]
Was replaced by https://github.com/theforeman/foreman-ansible-modules/pull/823

@jladdjr approved the changes here, and I think the important bits are still the same, so we may start pushing for that to get in.

**Pull request closed without merging**